src/pages/RFOX/hooks/useEpochHistoryQuery.ts

```diff
@@ -33,6 +35,11 @@ export const epochHistoryQueryFn = async (): Promise<EpochMetadata[]> => {
   while (startTimestamp < now) {
     const nextStartTimestamp = BigInt(dayjs.unix(Number(startTimestamp)).add(1, 'month').unix())
 
+    if (nextStartTimestamp >= now) {
+      // We cannot fetch blocks in the future using Etherscan's API, or we'll get an "Error! Block timestamp too far in the future" error
```

Comment:
Or maybe that's fine now without etherscan API 👀 see https://github.com/shapeshift/web/pull/7193